pkg/archive/archive.go

```diff
@@ -460,13 +460,19 @@ func (ta *tarAppender) addTarFile(path, name string) error {
 	//writing tar headers/files. We skip whiteout files because they were written
 	//by the kernel and already have proper ownership relative to the host
 	if !strings.HasPrefix(filepath.Base(hdr.Name), WhiteoutPrefix) && !ta.IDMappings.Empty() {
```

Comment:
No need to worry about whiteouts here. They don't apply to your use case and even when a custom converter is set when this function is used by overlay driver it will just get the uid from the header.